ffi/examples/read-table/arrow.c

```diff
@@ -187,28 +107,59 @@ static GArrowBooleanArray* slice_to_arrow_bool_array(const KernelBoolSlice slice
   return (GArrowBooleanArray*)ret;
 }
 
+static ExclusiveEngineData* apply_transform(
```

Comment:
nit: maybe some docs that this consumes the data and hands back a (potentially) new one